### ISSUE TITLE
Add support for multiremote in wdio testrunner

### ISSUE THREAD

**Issue by christian-bromann** [webdriverio/webdriverio#741]
- it should not be possible to mix remote and multiremote tests together (makes no sense to do so)

ref #737

**Comment by saulshanabrook** [comment]
From https://github.com/webdriverio/webdriverio/issues/737#issuecomment-135402481

> I agree with you, it should be a different wdio config and completely different tests.
> 
> I am just wondering if it is possible to use the wdio command line tool, with a wdio.conf.js file to support creating a multiremote, without having to import webdriverio and call multiremote()

**Comment by saulshanabrook** [comment]
Something like this:

```javascript
// wdio.conf.js
exports.config = {
    multiremote: {
        myChromeBrowser: {
            desiredCapabilities: {
                browserName: 'chrome'
            }
        },
        myFirefoxBrowser: {
            desiredCapabilities: {
                browserName: 'firefox'
            }
        }
    }
}
```

```javascript
// test.js
var firstBrowser = browsers.select('first')
var secondBrowser = browsers.select('second')

describe('App', function () {
  before('init', function *() {
    yield browsers.init()
  })
  after('end', function *() {
    yield browsers.end()
  })
  before('go to URL', function *() {
    yield browsers.url('/')
  })

  describe('Grandmaster level', function () {
    function * mustEqual (b, value) {
      yield b.getValue('#grandmaster .level input').must.eventually.equal(value.toString())
    }
    it('must start at 100', function *() {
      yield* mustEqual(firstBrowser, 100)
      yield* mustEqual(secondBrowser, 100)
      yield browsers.sync()
    })
    function * clickAndType (b, value) {
      yield b.waitForVisible('#grandmaster .level input')
      yield b.setValue('#grandmaster .level input', value.toString())
    }
    it('changing on one should change both', function *() {
      yield* clickAndType(firstBrowser, 10)
      yield* mustEqual(firstBrowser, 10)
      yield browsers.pause(2000)
      yield* mustEqual(secondBrowser, 10)

      yield browsers.sync()

      yield* clickAndType(secondBrowser, 20)
      yield* mustEqual(secondBrowser, 20)
      yield browsers.pause(2000)
      yield* mustEqual(firstBrowser, 20)
    })
  })
})
```

**Comment by christian-bromann** [comment]
Added support. Check out the [example config file](https://github.com/webdriverio/webdriverio/blob/master/examples/wdio/wdio.multiremote.conf.js) and the [example test file](https://github.com/webdriverio/webdriverio/blob/master/examples/wdio/runner-specs/mocha.multiremote.test.js) to check out how it works. It is pretty simple, instead of passing an array of capabilities you just assign the multiremote option object to the capability property. To keeps things simple I avoided to introduce a new config property. I make the decision whether or not it is a multiremote test based on if the capability property contains an array or not.

**Comment by saulshanabrook** [comment]
woot woot! looks great, thank you!

**Comment by christian-bromann** [comment]
Will publish version later today. I keep you posted.

**Comment by christian-bromann** [comment]
Released with `v3.2.1`
